src/unix/syscall.c

```diff
@@ -2333,6 +2332,20 @@ sysreturn umask(int mask)
     return mask;
 }
 
+sysreturn getcpu(unsigned int *cpu, unsigned int *node, void *tcache)
+{
+    cpuinfo ci = current_cpu();
+    if (!validate_user_memory(cpu, sizeof *cpu, true) ||
+            !validate_user_memory(node, sizeof *node, true))
+        return -EFAULT;
```

Comment:
Before calling validate_user_memory(), the pointers should be checked against a null value (otherwise -EFAULT would be returned when either `cpu` or `node` is NULL, which is a valid value)

Reply:
Good catch, thanks